### ISSUE TITLE
Domain is not pure kotlin code

### ISSUE THREAD

**Issue by rakib-bk** [Farhandroid/AndroidCleanArchitecture#12]
The domain class is using gson annotation. In clean architecture we write pure java/kotlin code in domain layer. The gson annotated class will remain in data layer and a mapper should be used to convert it from data layer to domain layer. Please correct me if I am wrong.

**Comment by HiImMatthewN** [comment]
You are correct, even the repository contains the 'PagingData' which is from the Android Paging Library itself.
